### PR TITLE
Added bookmarked-score to enable boosting of bookmarked items

### PR DESCRIPTION
**Pull request by adam-james-v**
Epic: #20318 
Covers: #20526 

Add scoring and proper queries in search to allow boosting of bookmarked cards, collections, and dashboards in search results.

WIP

### CODE REVIEW

File: src/metabase/api/search.clj [hunk 1]
```diff
@@ -272,6 +272,11 @@
 (s/defmethod search-query-for-model "collection"
   [model search-ctx :- SearchContext]
   (-> (base-query-for-model model search-ctx)
+      ;; TODO: get this working
```
Comment (dpsutton) [inline src/metabase/api/search.clj:275]: this comment needs to go. Either its already done or needs to be done.

**Review by dpsutton — APPROVED** [review]
looks good. nice test. Just confused about the `;;TODO: get this working` . probably just needs to be removed.

**Comment by dpsutton** [comment]
One other thing, I see that you added the bookmarked column for one of the models. It was already present for another one. Is it now on all three models that can be bookmarked? Hard to tell with this PR and surprising that they seem to have gone in at different times.

File: src/metabase/api/search.clj (hunk 1 above)
Comment (adam-james-v) [reply src/metabase/api/search.clj]: Removed the comment.

It's a bit weird, but the query is in search config:
https://github.com/metabase/metabase/pull/21171/files#diff-52b1b2f700169bc51d0f78761bfb84a70a7f1bdec8137889e2b089f30a22875eR158

**Pull request merged**